### PR TITLE
Implement loading and saving.

### LINKED ISSUE #39: Implement save & load

### PR DESCRIPTION
**Pull request by jeffswenson**
Resolves #39

### CODE REVIEW

File: src/duckling-tests/helpers/equality.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {ChangeType, changeType} from '../../duckling/state/object-diff';
+
+export function isDeepEqual(a : any, b : any) {
+    return changeType(a, b) === ChangeType.Equal;
```
Comment (jeffswenson) [inline src/duckling-tests/helpers/equality.ts]: I should delete this

File: src/duckling-tests/helpers/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './equality';
```
Comment (jeffswenson) [inline src/duckling-tests/helpers/index.ts]: I should delete this

**Comment by tlein** [comment]
When I first load in a map and then hit undo it goes back to the "Loading..." default screen for the app.

**Comment by tlein** [comment]
You should prettify the output

File: src/duckling/project/project.service.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            this._store.dispatch(doneLoadingProjectAction());
+        });
+    }
+
+    /**
+     * Save the projects current state.
+     */
+    save() {
+        var map = this._mapParser.systemToMap(MAP_NAME, this._entitySystem.entitySystem.value);
+        var json = JSON.stringify(map);
```
Comment (tlein) [inline src/duckling/project/project.service.ts]: You should prettify the JSON output.

File: src/duckling/project/map-parser.service.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            for (var systemKey in entity) {
+                if (!systems[systemKey]) {
+                    systems[systemKey] = {components: {}};
+                }
+                systems[systemKey].components[entityKey] = entity[systemKey];
+            }
+        });
+
+        return {
+            key : mapKey,
```
Comment (tlein) [inline src/duckling/project/map-parser.service.ts:79]: The key has ".map" at the end, this shouldn't be there.

File: src/duckling/project/project.service.ts (hunk 3 above)
Comment (jeffswenson) [inline src/duckling/project/project.service.ts]: Done

File: src/duckling/project/map-parser.service.ts (hunk 4 above)
Comment (jeffswenson) [inline src/duckling/project/map-parser.service.ts:79]: Done

**Pull request merged**